thunder/tests/opinfos.py

```diff
@@ -561,6 +561,24 @@ def _is_cuda_torch(x: torch.Tensor) -> bool:
 
 tensor_properties.append(is_cuda_opinfo)
 
+
+def logical_not_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    cases = (
+        (4, 4),
+        (5, 1, 3),
+    )
+    for input_shape in cases:
+        yield SampleInput(make(input_shape))
+
+
+logical_not_opinfo = OpInfo(
+    clang.logical_not,
+    sample_input_generator=logical_not_sample_generator,
+    torch_reference=torch.logical_not,
+)
+tensor_properties.append(logical_not_opinfo)
```

Comment:
`logical_not` is an elementwise unary operation:

https://github.com/k223kim/lightning-thunder/blob/8e1feeac38dfdaacab71e7557f480cd97a5b4deb/thunder/tests/opinfos.py#L439

they have their own OpInfos with their own generator, too

Reply:
Thanks for the information. Will add this to elementwise_unary_ops!